### PR TITLE
return Name objects instead of plain String

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndPullMessages.java [hunk 1]
```diff
@@ -52,7 +51,7 @@ public static void main(String... args) throws Exception {
         };
     Subscriber subscriber = null;
     try {
-      subscriber = Subscriber.Builder.newBuilder(subscription, receiver).build();
```
Comment (garrettjonesgoogle) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndPullMessages.java]: Thanks for this change, I kept forgetting to request it

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java [hunk 2]
```diff
@@ -121,6 +121,7 @@ public static long getApiMaxRequestBytes() {
 
   private static final Logger logger = LoggerFactory.getLogger(Publisher.class);
 
+  private final TopicName topicName;
   private final String topic;
```
Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java]: Maybe rename the String to "cachedTopicNameString" to indicate that the authoritative data source is topicName.

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Subscriber.java [hunk 3]
```diff
@@ -249,6 +263,7 @@ public void terminated(State from) {}
   private static class SubscriberImpl extends AbstractService {
     private static final Logger logger = LoggerFactory.getLogger(Subscriber.class);
 
+    private final SubscriptionName subscriptionName;
     private final String subscription;
```
Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Subscriber.java]: Maybe rename the String to "cachedSubscriptionNameString" to indicate that the authoritative data source is subscriptionName.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndPullMessages.java (hunk 1 above)
Comment (pongad) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndPullMessages.java]: :grinning:

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**